packages/supermassive/README.md

```diff
@@ -1,15 +1,46 @@
+Our goals are to provide a solution with the following optimizations for when all GraphQL operations (queries, mutations, subscriptions) needed by the user-experiences are statically known at build-time:
+
+- Bundle size of production targets needs to be as small as possible.
+- Performance is favored over runtime validation.
+
+Such is the case in our scenarios, where our schema lives entirely in the client application and some of these applications have very little to no GraphQL needs [yet] other than to suffice the needs of a single or few components.
+
+## Why
+
+Running a GraphQL executor can be an expensive exercise. The JavaScript community has the good fortune of having the official reference implementation of the GraphQL specification being implemented in JavaScript, namely [graphql-js](http://github.com/graphql/graphql-js). Inevitably this means that most general purpose GraphQL libraries in the JavaScript ecosystem end up wrapping it or otherwise rely on it. However, its goal is specifically to be an _all-encompassing_ implementation used for reference needs, _not_ to be an optimized solution for specific use-cases.
+
+## How
+
+Consider a GraphQL schema. It is typically a sizeable chunk of data, both in terms of type/field definition metadata as well as associated field-resolvers and _their_ code dependencies. Statically knowing all GraphQL operations allows us to reduce the bundle size to a minimum by [tree-shaking](https://en.wikipedia.org/wiki/Tree_shaking) all the definition metadata not required by any of the given operations. A JavaScript code bundler can in turn ensure only that code which is needed by the remaining field-resolvers is included in the production bundle. This means that the process is entirely dependency driven by needs expressed by the user-experiences, rather than requiring blunt manual configuration.
+
+Similarly, the GraphQL operations themselves, described using e.g. [GraphQL SDL](https://graphql.org/learn/schema/) or [`graphql-js` AST](https://github.com/graphql/graphql-js/blob/main/src/language/ast.ts), can incur quite some overhead as operations and number of operations grow. Eliminating these from the bundles can save size as well as runtime processing.
+
+## Roadmap
+
+### Phase 1
+
+In this initial phase, we will achieve the goal of tree-shaking the schema definitions. We do this by inlining required metadata into the documents that describe the operations, after which they can be executed with the need of the entire schema.
+
```

Comment:
In my head its a little hard to grasp the difference between this and the persisted queries step 3. I think it's effectively dropping bits of the schema used during dev (which you could show as a before/after)

Reply:
Doh, absolutely. That had been my very first want, and then forgot after writing all this 🤦

Reply:
Which do you think works better? I'm leaning towards the diff version:

<img width="998" alt="Screenshot 2021-12-08 at 16 55 26" src="https://user-images.githubusercontent.com/2320/145240942-516ff770-9ce7-40cf-b187-1df7691994ba.png">

<img width="989" alt="Screenshot 2021-12-08 at 17 00 05" src="https://user-images.githubusercontent.com/2320/145240958-ab7b59a0-c30c-484c-8568-8044443b2dd6.png">

Reply:
Going with the diff.